Chore: make timeseries the default panel in 8+

ryantxu opened the pull request:
In Grafana 8, we aim to replace the flot panel with the new uPlot based one.  We should make it the default ASAP ;)

![image](https://user-images.githubusercontent.com/705951/106971046-f69cd180-6702-11eb-812a-c3220c5f8794.png)

After we do more migrations work and better testing, we should aim to *deprecate* the graph panel in 7.5.

This PR also makes `barchart` beta and fiddles the presentation order

ryantxu committed:
make timeseries the default panel

ryantxu committed:
Chore: pad unknown values with undefined (#30808)

ryantxu committed:
update order

leeoniya commented:
until we have better parity with the graph panel (#30564: stacking, thresholds, annotation editing), we should at least warn people about stuff that is still missing so they don't end up setting up a lot of stuff and then realizing there's no reverse-migration path and have to redo their whole config in a blinding rage.

this plays nicely into the whole "recommend the appropriate viz/panel" on our roadmap.

ryantxu committed:
add panel edit banner

ryantxu committed:
add panel edit banner

ryantxu commented:
I'm torn -- I think the sooner people use the new goodness the better.  Then they don't have to risk migrations later.  but yes there are a couple things that are not at parity yet.  

Maybe adding a banner like this:
![image](https://user-images.githubusercontent.com/705951/106996427-d5090d80-6735-11eb-9fe7-903132c77cda.png)

And lets aim to remove it before 7.5 release?

ryantxu commented on packages/grafana-data/src/panel/PanelPlugin.ts:
```diff
@@ -106,6 +106,7 @@ export class PanelPlugin<
   onPanelMigration?: PanelMigrationHandler<TOptions>;
   onPanelTypeChanged?: PanelTypeChangedHandler<TOptions>;
   noPadding?: boolean;
+  editBanner?: ComponentType | null;
```
or maybe better to hardcode for `timeseries' and 'graph' migration issues

hugohaggmark committed:
Variables: Adds queryparam formatting option (#30858)

* Variables: Adds queryparam formatting option

* Chore: fixes strict errors

* Chore: changes after PR comments

torkelo commented:
The biggest potential blocker for doing this is alerting

torkelo committed:
Alerts: Update Alert storybook to show more states (#30908)

* Alert: Design update for alerts

* Before text color

* Text update

* reverted style changes

* Minor fix

* Updated snapshot

thisisobate committed:
Docs: updated developer guide (#29978)

* Docs: updated developer guide

* Fix: made heading uppercase

* small nit fix

marefr committed:
Chore: Upgrade docker build image (#30820)

* Chore: Upgrade docker build image

* use updated build image in drone

hugohaggmark committed:
e2e: extends selector factory to plugins (#30932)

Commit:
Docs: Add alias for what's new in 7.4 (#30945)

torkelo committed:
Theme: Use higher order theme color variables rather then is light/dark logic (#30939)

torkelo committed:
Table: Updates devenv test dashboard after change to TestData Randrom Table response (#30927)

sunker committed:
assign changes in cloud datasources to the new cloud datasources team (#30645)

sunker committed:
make sure service and slo display name is passed to segment comp (#30900)

torkelo committed:
Alerts: Dedupe alerts so that we do not fill the screen with the same alert messsage (#30935)

Commit:
Elasticsearch: Show Size setting for raw_data metric (#30980)

torkelo committed:
Units: Fixes formatting of duration units (#30982)

dependabot[bot] committed:
Chore(deps): Bump actions/cache from v2 to v2.1.4 (#30973)

Bumps [actions/cache](https://github.com/actions/cache) from v2 to v2.1.4.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...26968a09c0ea4f3e233fdddbafd1166051a095f6)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

torkelo committed:
StatPanel: Fixes issue formatting date values using unit option (#30979)

oscarkilhed committed:
Grafana/UI: Remove DismissableFeatureInfoBox and replace with LocalSt… (#30988)

* Grafana/UI: Remove DismissableFeatureInfoBox and replace with LocalStorageValueProvider

bergquist committed:
instrumentation: make the first database histogram bucket smaller (#30995)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

weisdd committed:
Docs: Update configuration.md (login_maximum_inactive_lifetime_duration, login_maximum_lifetime_duration) (#31000)

Grafana [v7.2.0-beta2](https://github.com/grafana/grafana/releases/tag/v7.2.0-beta2) brought support for `login_maximum_inactive_lifetime_duration` and `login_maximum_lifetime_duration` to replace the older way of specifying the same flavor of timeouts (`login_maximum_inactive_lifetime_days` and `login_maximum_lifetime_days`). Unfortunately, this improvement did not make it to `configuration.md`.

The suggested changes reflect the current state of `conf/defaults.ini`.

hugohaggmark committed:
TextPanel: Fixes so panel title is updated when variables change (#30884)

* TextPanel: Fixes so panel title is updated when variables change

* Tests: fixes tests

* Chore: updates after PR comments

hugohaggmark committed:
Build: Releases e2e and e2e-selectors too (#31006)

dependabot[bot] committed:
Chore(deps): Bump google.golang.org/api from 0.33.0 to 0.39.0 (#30971)

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.33.0 to 0.39.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/master/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.33.0...v0.39.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sunker committed:
CloudMonitoring: Prevent resource type variable function from crashing (#30901)

* check if label response has resource type prop

* use nullish coalescing operator

hugohaggmark committed:
Graph: Fixes so graph is shown for non numeric time values (#30972)

* Graph: Fixes so graph is shown for non numeric time values

* Tests: changes times to UTC

* Tests: forgot one value in time array

* GraphNG: make time series panel work with string time stamps (#30981)

* Make Time series panel work with data that time is represented as string

* Map to for

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

dependabot[bot] committed:
Chore(deps): Bump github.com/grpc-ecosystem/go-grpc-middleware (#31013)

Bumps [github.com/grpc-ecosystem/go-grpc-middleware](https://github.com/grpc-ecosystem/go-grpc-middleware) from 1.2.1 to 1.2.2.
- [Release notes](https://github.com/grpc-ecosystem/go-grpc-middleware/releases)
- [Changelog](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/CHANGELOG.md)
- [Commits](https://github.com/grpc-ecosystem/go-grpc-middleware/compare/v1.2.1...v1.2.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump github.com/magefile/mage from 1.10.0 to 1.11.0 (#31017)

Bumps [github.com/magefile/mage](https://github.com/magefile/mage) from 1.10.0 to 1.11.0.
- [Release notes](https://github.com/magefile/mage/releases)
- [Changelog](https://github.com/magefile/mage/blob/master/.goreleaser.yml)
- [Commits](https://github.com/magefile/mage/compare/v1.10.0...v1.11.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nutmos committed:
Prometheus: Min step defaults to seconds when no unit is set to prevent errors when running alerts. (#30966)

* update interval to fix no unit

* fix bug

* update with new method

* format go

* add test

* change test method to add test error nil

* add simplejson and model package

* change json format

* add parentheses

* add simplejson to function

* change mode to model

* move assert import package

* add one enter to avoid goimport error

* change to test case

dependabot[bot] committed:
Chore(deps): Bump github.com/aws/aws-sdk-go from 1.36.31 to 1.37.7 (#31018)

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.36.31 to 1.37.7.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.36.31...v1.37.7)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump github.com/jung-kurt/gofpdf from 1.10.1 to 1.16.2 (#30586)

Bumps [github.com/jung-kurt/gofpdf](https://github.com/jung-kurt/gofpdf) from 1.10.1 to 1.16.2.
- [Release notes](https://github.com/jung-kurt/gofpdf/releases)
- [Commits](https://github.com/jung-kurt/gofpdf/compare/v1.10.1...v1.16.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jackw committed:
Dashboard: Migrate general settings to react (#30914)

* feat(dashboard): initial commit of general settings migration to react

* fix(dashboardsettings): force update of general settings when selects change

* feat(dashboardsettings): initial commit of delete dashboard button and modal

* feat(dashboardsettings): introduce useDashboardDelete hook

* feat(dashboardsettings): add tags and editable inputs

* refactor(dashboardsettings): fix typescript error in general settings

* refactor(dashboardsettings): use grafana-ui form components for general settings

* refactor(dashboardsettings): use ConfirmModal and move provisioned modal to own component

* refactor(dashboardsettings): revertDashboardModal uses ConfirmModal

* test(autorefreshintervals): remove renderCount prop to fix test

* test(dashboardsettings): put back aria-label for e2e tests

* chore(dashboardsettings): remove redundant generl settings angular code

* test: change references to now deleted SettingsCtrl to GeneralSettings

* refactor(dashboardsettings): swap out switch for inlineswitch component

* chore(timepickersettings): remove timePickerSettings angular directive definition

* feat(dashboardsettings): add tooltips, fix description field name

* refactor(dashboardsettings): remove redundant await

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* refactor(usedashboarddelete): clean up

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

thisisobate committed:
ColorPicker: migrated styles from sass to emotion (#30909)

* ColorPicker: migrated styles from sass to emotion

* fixes frontend test

* updated some changes

* updated some changes

* fixed small import nits

aknuds1 committed:
Drone: Fix deployment image (#31027)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
BarGauge: Improvements to value sizing and table inner width calculations  (#30990)

* BarGauge: Increase min value width and fix height when setting manual text size

* updated snapshot

* Big improvement to bar gauge value sizing and fixing table gauge sizing

* removed unused const

* added a unit test

carriecrowe1138 committed:
Improved glossary (#31004)

torkelo committed:
CDN: Adds uppload to CDN step to drone CI (#30879)

* CDN: Adds uppload to CDN step to drone CI

* Bump grabpl

* updated sh files

* crying

* add init dependency

* Removed last test to get this to work

Commit:
Elasticsearch: Fix alias field value not being shown in query editor (#30992)

* Make imports relative

* Fix Alias field value not being shown

ivanahuckova committed:
Explore: Remove emotion error when displaying logs (#31026)

* Remove emotion error

* Remove redundant context variable

hugohaggmark committed:
Alerting: Fixes so notification channels are properly deleted (#31040)

hugohaggmark committed:
DashboardLinks: Fixes crash when link has no title (#31008)

* DashboardLinks: Fixes crash when link misses title

* Chore: updates after PR comments

wbrowne committed:
convert path to posix by default (#31045)

torkelo committed:
CDN: Fixing drone CI config (#31052)

* CDN: Fixing drone CI config

* was in wrong dir when I executed command

* removed whitespace

torkelo committed:
DashboardSettings: Update to new form styles (#31022)

* DashboardSettings: Update to new form styles

* Improve header margin and fixing issue in permissions tab

* Updated

* Switch to radio

* Remove unused import

achatterjee-grafana committed:
Added hyperlinks to Graphite documentation (#31064)

* Added references to Graphite doc pages.

* Minor tweak.

* Fixed a typo.

oddlittlebird committed:
Docs: Update whats-new-in-v7-4.md (#31069)

oddlittlebird committed:
Docs: Update json-model.md (#31066)

sunker committed:
Chore: Replace native select with grafana ui select  (#31030)

* use react select instead of simple select

* cleanup

* rename selectable value prop

* remove not used import

torkelo committed:
CDN: Add CDN upload step to enterprise and release pipelines (#31058)

* CDN: Fixing drone CI config

* was in wrong dir when I executed command

* removed whitespace

* Update path

* Update Drone master and enterprise piplelines with CDN Upload

* Update test

* removed whitespace

* Go back to public folder defined

* updated build-pipeline

* Updated drone

* wrong version

bergquist committed:
Usage stats: Adds source/distributor setting (#31039)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

marefr committed:
Docs: Group id should be 0 instead of 1 in Docker upgrade notes (#31074)

The upgrade notes for Grafana 7.3 and Docker says that The Grafana Docker 
image runs with the root group (id 1) which is wrong. It should be group (id 0).

Which issue(s) this PR fixes:
Fixes #30953

jcbize committed:
Transformations: Fixed typo in FilterByValue transformer description. (#31078)

vickyyyyyyy committed:
test: update addDashboard flow for v7.4.0 changes (#31059)

* update addDashboard flow for v7.4.0 changes

* remove hide flow and if around set time range

aknuds1 committed:
PluginManager: Remove some global state (#31081)

* PluginManager: Remove global state

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Drone: Make sure CDN upload is ok before pushing docker images (#31075)

* Drone: Make sure CDN upload is ok before pushing docker images

* Trying to simplify logic

* Snake case

* Only depend on cdn upload in release mode

* Rename and use util function where we can

* removing the cdn dependency

ivanahuckova committed:
LogsPanel: Add deduplication option for logs (#31019)

* WIP: Add dedup functionality to logs panel

* Simplify, clean up

* Change ordering of customizations based on how it is in Explore

* Update public/app/plugins/panel/logs/LogsPanel.tsx

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

* Update public/app/plugins/panel/logs/LogsPanel.tsx

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

ivanahuckova committed:
Prometheus: Fix issues with ad-hoc filters (#30931)

* Don't add {} to non-metrics starting with quotes

* Fix for using template variables

* Fix adding of filter to timerange

* Fix adding of filter to labels provided with group modifier

* Fix adding of filters to Grafana's variables

* Shorten tests

* Refactor and split

Commit:
Elasticsearch: fix log row context erroring out (#31088)

vickyyyyyyy committed:
update configurePanel for 7.4.0 changes (#31093)

xlson committed:
Docs: request security (#30937)

* Docs: request security

* Docs: lists the section in all examples

* typo

* Update docs/sources/enterprise/_index.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/enterprise-configuration.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Joan López de la Franca Beltran <joanjan14@gmail.com>

* Update docs/sources/enterprise/enterprise-configuration.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/enterprise-configuration.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/enterprise-configuration.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* whats new

* Update docs/sources/enterprise/request-security.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update docs/sources/enterprise/request-security.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update docs/sources/whatsnew/whats-new-in-v7-4.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* final edits

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>
Co-authored-by: Joan López de la Franca Beltran <joanjan14@gmail.com>
Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>
Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

aocenas committed:
Influx: Make max series limit configurable and show the limiting message if applied (#31025)

* Add configuration in ConfigEditor and default to 1000

* Show data in explore if any even if there is an error

* Update pkg/tsdb/influxdb/flux/executor.go

* Better handling of defaults

* Add test for runQuery to show data even with error

* Update public/app/store/configureStore.ts

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

* Update public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* Update tooltip

* Update input

* Lint fixes

* Update snapshots

* Update decorator tests

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>
Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

zoltanbedi committed:
Exemplars: Change CTA style (#30880)

* Exemplars: Change CTA style

* Address review feedbacks

* Fix table column aligning

* Minor alignments + uncontrolled component warning fix

briangann committed:
area/grafana/e2e: ginstall should pull version specified (#31056)

* fall back to version specified
* update docker image and bump version
* update go version, linter, and codeclimate

AgnesToulet committed:
Search: add sort information in dashboard results (#30609)

* Search: add SortMeta in dashboard results

* fix integration tests

* trim SortMeta

* fix searchstore tests

* Update pkg/services/sqlstore/dashboard.go

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

domasx2 committed:
Logging: sourcemap transform asset urls from CDN in logged stacktraces (#31115)

aknuds1 committed:
MuxWriter: Handle error for already closed file (#31119)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Transforms: Fixes Outer join issue with duplicate field names not getting the same unique field names as before  (#31121)

* Transformations: Fixed duplicate name issue in outer join transform

* Think this is working

* Updated tests

* Updated snapshot

* Fix broken tests (#31123)

* Fix broken tests

* Fix remaining faling tests

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

dprokop committed:
Eslint: no-duplicate-imports rule (bump grafana-eslint-config) (#30989)

* Eslint: no-duplicate-imports rule (bump grafana-eslint-config)

* Chore: Fix duplicate imports (#31041)

* Rebased this branch into eslint-no-duplicate-imports

* updated some changes

* merged uncaught duplicate imports

* fixes frontend test- I hope

* fixes e2e test- I hope

Co-authored-by: Uchechukwu Obasi <obasiuche62@gmail.com>

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.4.1 (#31133)

torkelo committed:
PieChart: Progress on new core pie chart  (#28020)

Use visx for the Pie chart plugin.

Co-authored-by: Oscar Kilhed <oscar.kilhed@grafana.com>

Commit:
Docs: add 7.4.1 relese notes link (#31137)

Commit:
Chore: Update latest.json (#31139)

kurokochin committed:
Alerting: Fix so that sending an alert with the Alertmanager notifier doesn't fail when one of multiple configured URL's are down (#31079)

Fixes behaviour of Notify that returns error when one of the dispatch 
event return error, to maintain high availability, we should return error 
when all dispatched events return error instead.

Fixes #30509

torkelo committed:
DataSourceSettings: Fixes add header button, it should not trigger a save & test action (#31135)

ivanahuckova committed:
Prometheus: Add documentation for ad-hoc filters (#31122)

* Prometheus: Add documentation for ad-hoc filters

* Update docs/sources/datasources/prometheus.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/datasources/prometheus.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Add link to ad hoc filters doumentation

* Change direct url for relative

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

dsotirakis committed:
Add author name and pr number in drone pipeline notifications (#31124)

* Add author name and pr number in drone pipeline notifications

torkelo committed:
DashboardListPanel: Fixes issue with folder picker always showing All and using old form styles (#31160)

zoltanbedi committed:
Prometheus: Multiply exemplars timestamp to follow api change (#31143)

Clarity-89 committed:
Variables: Extend option pickers to accept custom onChange callback (#30913)

* Rename DashboardPicker props

* Extend variable option pickers to allow providing custom onchage callback

* Update public/app/features/variables/pickers/OptionsPicker/actions.ts

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

* Update public/app/features/variables/pickers/OptionsPicker/actions.ts

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

* Update public/app/features/variables/pickers/OptionsPicker/actions.ts

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

* Update public/app/features/variables/textbox/TextBoxVariablePicker.tsx

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

* Codeformat

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

zoltanbedi committed:
Update prometheus.md (#31173)

Exemplars won't make it to the 2.25 release.

sunker committed:
CloudWatch: Ensure empty query row errors are not passed to the panel (#31172)

* ensure empty errors are not passed to the panel

* make it a oneliner

AgnesToulet committed:
Permissions: Fix team and role permissions on folders/dashboards not displayed for non Grafana Admin users (#31132)

* Cfg: fix hidden users initialization

* add tests

* do not call isHiddenUser function for non-user permission

* do not call isHiddenUser function for non-user permission

thisisobate committed:
Alert: update story to use controls (#31145)

* Alert: updated story to use Controls

* updated some changes

dsotirakis committed:
Replace PR with Commit truncated hash when build fails (#31177)

* Replace PR with Commit truncated hash when build fails

* Make commit literal lower case

torkelo committed:
DashboardLinks: Fixes links always cause full page reload (#31178)

aocenas committed:
Explore/Refactor: Simplify URL handling (#29173)

* Inline datasource actions into initialisation

* Simplify url handling

* Add comments

* Remove split property from state and split Explore.tsx to 2 components

* Add comments

* Simplify and fix splitOpen and splitClose actions

* Update public/app/features/explore/ExplorePaneContainer.tsx

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

* Update public/app/features/explore/state/explorePane.test.ts

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>

* Update public/app/features/explore/Wrapper.tsx

Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

* Fix test

* Fix lint

Co-authored-by: Giordano Ricci <gio.ricci@grafana.com>
Co-authored-by: Ivana Huckova <30407135+ivanahuckova@users.noreply.github.com>

oddlittlebird committed:
Docs: Rewrite preferences docs (#31154)

* Create _index.md

* Moved file and started to update content

* adding content

* Update change-home-dashboard.md

* Update _index.md

* Update _index.md

* Update _index.md

* content edits and moved file

* Update change-grafana-name.md

* Delete preferences.md

* content changes

* content changes

* content updates

* content updates

* Update docs/sources/administration/preferences/_index.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/_index.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/_index.md

* Update docs/sources/administration/preferences/_index.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/_index.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/_index.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/change-grafana-name.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/change-grafana-name.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/change-grafana-name.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update docs/sources/administration/preferences/change-grafana-theme.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

* Update _index.md

* Update _index.md

* minor edits

* Update change-grafana-name.md

Co-authored-by: Ursula Kallio <73951760+osg-grafana@users.noreply.github.com>
Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

achatterjee-grafana committed:
Fixed the typo. (#31189)

torkelo committed:
StatPanels: Fixes to palette color scheme is not cleared when loading panel (#31126)

hugohaggmark committed:
Variables: Fixes missing empty elements from regex filters (#31156)

* Variables: Fixes missing empty elements from regex filters

* Chore: cleanup comment

* Chore: removes unused import

dependabot[bot] committed:
Chore(deps): Bump github.com/linkedin/goavro/v2 from 2.9.7 to 2.10.0 (#31204)

Bumps [github.com/linkedin/goavro/v2](https://github.com/linkedin/goavro) from 2.9.7 to 2.10.0.
- [Release notes](https://github.com/linkedin/goavro/releases)
- [Changelog](https://github.com/linkedin/goavro/blob/master/debug_release.go)
- [Commits](https://github.com/linkedin/goavro/compare/v2.9.7...v2.10.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

papagian committed:
Alerting: Fix modal text for deleting obsolete notifier (#31171)

dependabot[bot] committed:
Chore(deps): Bump xorm.io/xorm from 0.8.1 to 0.8.2 (#30773)

* Chore(deps): Bump xorm.io/xorm from 0.8.1 to 0.8.2

Bumps xorm.io/xorm from 0.8.1 to 0.8.2.

Signed-off-by: dependabot[bot] <support@github.com>

* Fix limit for snapshots

* Fix limit for org and users

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

dependabot[bot] committed:
Chore(deps): Bump cloud.google.com/go/storage from 1.12.0 to 1.13.0 (#31211)

Bumps [cloud.google.com/go/storage](https://github.com/googleapis/google-cloud-go) from 1.12.0 to 1.13.0.
- [Release notes](https://github.com/googleapis/google-cloud-go/releases)
- [Changelog](https://github.com/googleapis/google-cloud-go/blob/master/CHANGES.md)
- [Commits](https://github.com/googleapis/google-cloud-go/compare/spanner/v1.12.0...spanner/v1.13.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

joshhunt committed:
Chore: report eslint no-explicit-any errors to metrics (#31182)

* Chore: report eslint no-explicit-any errors to metrics

* double quote

torkelo committed:
QueryEditors: Fixes issue that happens after moving queries then editing would update other queries (#31193)

torkelo committed:
DataSourceSrv: Filter out non queryable data sources by default (#31144)

wbrowne committed:
use 0.1.0 (#31215)

dafydd-t committed:
WIP: Skip this call when we skip migrations (#31216)

aocenas committed:
Docs: Update Influx config options (#31146)

* Add config options

* Update docs/sources/datasources/influxdb.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update docs/sources/datasources/influxdb.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Add config to provisioning docs

* Style fixes

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

peterholmberg committed:
AlertingNG:  Test definition  (#30886)

* break out new and edit

* changed model to match new model in backend

* AlertingNG: API modifications (#30683)

* Fix API consistency

* Change eval alert definition to POST request

* Fix eval endpoint to accept custom now parameter

* Change JSON input property for create/update endpoints

* model adjustments

* set mixed datasource, fix put url

* update snapshots

* run test response through converters

* remove edit and add landing page

* remove snapshot tests ans snapshots

* wrap linkbutton in array

* different approaches to massage data

* get instead of post

* use function to return instances data

* hook up test button in view

* test endpoint for not saved definitions

* function that return query options

* Chore: fixes strict error

* hide ng alert button

* typings

* fix setAlertDef error

* better message when you have queries but no data

* NGAlert: Refactoring that handles cleaning up state (#31087)

* Chore: some refactorings of state

* Chore: reduces strict null errors

Co-authored-by: Sofia Papagiannaki <papagian@users.noreply.github.com>
Co-authored-by: Sofia Papagiannaki <sofia@grafana.com>
Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>
Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

torkelo committed:
Graph: Make axes unit option work even when field option unit is set (#31205)

* Graph: Make axes unit option work even when field option unit is set

* Updated tests

marefr committed:
Chore: Upgrade grafana-plugin-sdk-go to v0.88.0 (#30975)

* Chore: Upgrade grafana-plugin-sdk-go

* update plugin sdk

Co-authored-by: Will Browne <will.browne@grafana.com>

undef1nd committed:
Remove last synchronisation field from LDAP debug view (#30984)

* Remove last synchronisation field from LDAP debug view

* Apply review comments

dprokop committed:
GraphNG: refactor core to class component (#30941)

* First attempt

* Get rid of time range as config invalidation dependency

* GraphNG class refactor

* Get rid of DataFrame dependency from Plot component, get rid of usePlotData context, rely on XYMatchers for data inspection from within plugins

* Bring back legend

* Fix Sparkline

* Fix Sparkline

* Sparkline update

* Explore update

* fix

* BarChart refactor to class

* Tweaks

* TS fix

* Fix tests

* Tests

* Update packages/grafana-ui/src/components/uPlot/utils.ts

* Update public/app/plugins/panel/timeseries/plugins/ContextMenuPlugin.tsx

* GraphNG: unified legend for BarChart, GraphNG & other uPlot based visualizations (#31175)

* Legend experiment

* Nits

kurokochin committed:
Postgres: Fix timeGroup macro converts long intervals to invalid numbers when TimescaleDB is enabled (#31179)

Fixes #27253

aknuds1 committed:
IPv6: Support host address configured with enclosing square brackets (#31226)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

hugohaggmark committed:
Variables: Adds back default option for data source variable (#31208)

* Variables: Adds back default option for data source variable

* Chore: updates after PR comments

torkelo committed:
TagsInput: Design update and component refactor (#31163)

* TagsInput: Design update and component refactor

* Update packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Update packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Update packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Update packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Updated

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

dupondje committed:
Disable Change Password for OAuth users (#27886)

hugohaggmark committed:
LibraryPanels: Disconnect before connect during dashboard save (#31235)

* LibraryPanels: Disconnect before connect during dashboard save

* Tests: fixed test

* Chore: updates after PR comments

* Chore: changes from context.Background() to c.Context.Req.Context()

* Chore: fixes lint issue

torkelo committed:
SqlDataSources: Fixes the Show Generated SQL button in query editors (#31236)

* SqlDataSources: Fixes the Show Generated SQL button in query editors

* No need to protect against duplicate events now that the event emitter is isolated for each editor

ivanahuckova committed:
Remove deprecated and breaking loki config field (#31227)

ivanahuckova committed:
Loki: Add line limit for annotations (#31183)

* Add line limit to Annotation query editor

* Refactor to keep type consistent

kaydelaney committed:
UI/Card: Fix handling of 'onClick' callback (#31225)

ivanahuckova committed:
Prometheus: Fix enabling of disabled queries when editing in dashboard (#31055)

* Fix disable bug by passing hide prop

* Make more universal fix

* Revert to original fix

* Update public/app/plugins/datasource/prometheus/components/PromQueryEditor.tsx

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

joshhunt committed:
Grafana-UI: Add id to Select to make it easier to test (#31230)

* Prettier formatting

* Grafana-UI: Add support for id and inputId props to Select

* Grafana-UI: Add aria-label to Select

* Grafana-UI: InlineField and Field get ID from inputId prop

For Select

* Fix tests using TagFilter

* Update Select prop documentation

* Update Field tests to use screen instead

* Fix the last few tests

jackw committed:
Chore: grafana-toolkit uses grafana-ui and grafana-data workspaces (#30701)

* chore(grafana-toolkit): use workspace versions of grafana/ui and grafana/data

* chore: replace references to popperjs 1 typings with popperjs 2 typings

bergquist committed:
only update usagestats every 30min (#31131)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

marefr committed:
Snapshots: Disallow anonymous user to create snapshots (#31263)

sakjur committed:
UsageStats: Purpose named variables (#31264)

dprokop committed:
Make Datetime local (No date if today) working (#31274)

torkelo committed:
Folders: Editors should be able to edit name and delete folders (#31242)

* Folders: Editors should be able to edit name and delete folders

* Fixed so that we only build nav model in one place

Clarity-89 committed:
Search: display sort metadata (#31167)

* Search: display metadata

* Search: update SortPicker icon

* Search: display folder meta data

* Search: reset sort picker on layout change

* Search: align tags in Card component

* Search: replace hyphen with dash

* Search: preserve sort state on layout change

* Search: update tests

* Search: fix tests

* Update pkg/services/search/hits.go

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update public/app/features/search/components/SearchItem.tsx

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update public/app/features/search/components/SearchItem.tsx

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Update public/app/features/search/types.ts

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

* Search: fix type error

* Search: add General folder name and adjust icon margin

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

kaydelaney committed:
Introduce functions for interacting with library panels API (#30993)

torkelo committed:
Docs: Annotations update (#31194)

Fixes #31188

thisisobate committed:
BigValue: updates story from knobs to controls (#31240)

* BigValue: updates story from knobs to controls

* some refactoring to make code more readable

* made some changes

hugohaggmark commented on public/app/plugins/panel/timeseries/EditBanner.tsx:
```diff
@@ -0,0 +1,28 @@
+    >
+      <p>
+        The new time series panel is now the default panel in grafana. For most things, this is will be the best choice.
+        However a few features exist in the graph panel that are not yet implemented in this panel. Speciffically:
+        <ul>
+          <li>Annotaiton editor</li>
```
nit: small typos
```suggestion
        However, a few features exist in the graph panel that are not yet implemented in this panel. Specifically:
        <ul>
          <li>Annotation editor</li>
```

hugohaggmark reviewed (commented):
Found some typos

hugohaggmark committed:
Docs: Clarifies custom date formats for variables (#31271)

LeviHarrison committed:
Correct name of Discord notifier tests (#31277)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.4.2 (#31291)

ying-jeanne committed:
update latest release version (#31296)

hugohaggmark committed:
Chore: reduce strict errors for variables (#31241)

* Chore: reduces a lot of variable errors

* Chore: reduces variable Editor errors

* Chore: reduces variable Picker errors

* Chore: reduce error count

* Chore: reduces errors for ChangeEvent instead of FormEvent

* Chore: reduces errors with CombinedState

* Chore: reduces ComponentType errors

* Chore: reduce errors in reducers

* Chore: reduces misc errors

* Chore: reduce AdhocPicker errors

* Chore: reduce error limit

* Update public/app/features/variables/adhoc/picker/AdHocFilterValue.tsx

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

* Chore: updates after PR comments

* Chore: small refactor

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

dsotirakis committed:
Add `--tries 3` arg when triggering e2e-tests upon releasing (#31285)

* Add ``--tries 3` arg when triggering e2e-tests

* Make e2e-tests retry 3 times only for release pipelines

ying-jeanne committed:
fix 7.4.2 release note (#31299)

Clarity-89 committed:
Grafana-UI: Update tooltip type (#31310)

vickyyyyyyy committed:
test: allow check for Table as well as Graph for Explore e2e flow (#31290)

* use table for explore flow screenshot

* reverse logic

* update matchExploreTable type

thisisobate committed:
ButtonCascader: updates story from knobs to controls (#31288)

* ButtonCascader: updates story from knobs to controls

* some nit fixes

bergquist committed:
Usagestats: Exclude folders from total dashboard count (#31320)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

hugohaggmark committed:
Docs: Clarifies how to add Key/Value pairs (#31303)

thisisobate committed:
BarGuage: updates story from knobs to controls (#31223)

* BarGuage: updated story from knobs to controls

* fixes type errors and disables unused controls

* update controls to support some BarGauge props

* changed addBarGaugeStory to component and removed closure

* cleaned up to be more readable

* more refactoring to be readable

* final touches...

* fixed disabled controls issue

* moved NOOP_CONTROL declaration to a seperate PR

* made some tweaks to the story names

torkelo committed:
Table: Fixes issue with fixed min and auto max with bar gauge cell (#31316)

Commit:
Add info to docs about concurrent session limits (#31333)

* Add info to docs about concurrent session limits

This parallels a [change in the Grafana Cloud docs](https://github.com/grafana/website/pull/3504) which was based on a kind and appreciated message from support. The limit does not apply to Grafana OSS, just to Enterprise and Cloud, as far as any of us know.

* Update docs/sources/enterprise/license-restrictions.md

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

Co-authored-by: achatterjee-grafana <70489351+achatterjee-grafana@users.noreply.github.com>

aknuds1 committed:
Chore: Upgrade golangci-lint (#31330)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

hugohaggmark committed:
LibraryPanels: Syncs panel title with name (#31311)

torkelo committed:
DashboardLinks: Fixes another issue where dashboard links cause full page reload (#31334)

gabor committed:
QueryEditor: handle query.hide changes in angular based query-editors (#31336)

* handle query.hide changes in angular query-editors

* better comment

xlson committed:
SAML: single logout only enabled in enterprise (#31325)

oscarkilhed committed:
Grafana/UI: Add basic legend to the PieChart (#31278)

* Add basic legend to the PieChart

* Remove console log

* Remove PieChartWithLegend and refactor PieChart a bit

* Use FALLBACK_COLOR

* Refactor PieChart

* Change back to [function]

torkelo committed:
TimeSeriesPanel: Fixes overlapping time axis ticks (#31332)

* GraphNG: Fixes overlapping time axis ticks

* Review tweaks

dprokop committed:
DatasourceSrv: Fix instance retrieval when datasource variable value set to "default" (#31347)

* Failing tests

* Fixed

dprokop committed:
Update grot template (needs more info) (#31350)

zoltanbedi committed:
Zipkin: Show success on test data source (#30829)

* Zipkin: Show success on test data source

Add zipkin example app to devenv
Migrate to fetch api in data source

* Add back return types for functions

ivanahuckova committed:
Devenv: Remove old-versioned loki blocks and update prometheus2 block  (#31282)

* Remove very old-versioned loki blocks

* Update prometheus block from 2.7.2 to latest

* Remove data sources in Grafana

* Fix e2e test not to intercept the whole url

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

torkelo committed:
Theming: Support for runtime theme switching and hooks for custom themes (#31301)

* WIP Custom themes

* Load custom themes from URL and via event

* Dynamic page background

* Header color change

* Fixing tests and emotion warnings

* Fixed test

* moving cx to getStyles

* Review fixes

* minor change

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 7.4.2 (#31313)

hugohaggmark committed:
StoryBook: Introduces Grafana Controls (#31351)

dependabot[bot] committed:
Chore(deps): Bump github.com/getsentry/sentry-go from 0.9.0 to 0.10.0 (#31378)

Bumps [github.com/getsentry/sentry-go](https://github.com/getsentry/sentry-go) from 0.9.0 to 0.10.0.
- [Release notes](https://github.com/getsentry/sentry-go/releases)
- [Changelog](https://github.com/getsentry/sentry-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-go/compare/v0.9.0...v0.10.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

torkelo committed:
Chore: Fix strict errors, down to 416 (#31365)

dependabot[bot] committed:
Chore(deps): Bump github.com/golang/mock from 1.4.4 to 1.5.0 (#31379)

Bumps [github.com/golang/mock](https://github.com/golang/mock) from 1.4.4 to 1.5.0.
- [Release notes](https://github.com/golang/mock/releases)
- [Changelog](https://github.com/golang/mock/blob/master/.goreleaser.yml)
- [Commits](https://github.com/golang/mock/compare/v1.4.4...v1.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hugohaggmark committed:
AdHocVariables: Fixes crash when values are stored as numbers (#31382)

dependabot[bot] committed:
Chore(deps): Bump github.com/prometheus/common from 0.15.0 to 0.17.0 (#31387)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.15.0 to 0.17.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.15.0...v0.17.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hugohaggmark committed:
Revert "StoryBook: Introduces Grafana Controls (#31351)" (#31388)

This reverts commit c5797cc2748829f7f609c180044c8211d63f61b7.

aknuds1 committed:
Chore: Remove gotest.tools dependency (#31391)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Clarity-89 committed:
Data source list: Use Card component (#31326)

* Replace DataSourcesListItem with Card

* Add tests

* Remove unused styles

* Make card heading semi bold

* Make heading semi-bold

* Show type name instead of type id

* Fix key warning

* Update Card

* Fix tests

* Make typeName optional

* remove styling that was just a test

* Make typeName non-optional and fix tests

* Update list key

Co-authored-by: Torkel Ödegaard <torkel@grafana.com>

ivanahuckova committed:
Prometheus: Change default httpMethod for new instances to POST (#31292)

* Implement POST to specific endpoints and change POST as default

* Add tests for PromSettings

* Add tests to dataosurce.ts for new functionality

* Remove /label /series implementation

* Update public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx

achatterjee-grafana committed:
Added section Query a time series database by id (#31337)

* Added new section Query a time series database by id

* Rearranged content.

* changed database to data source

* Updated note with the word Grafana

* fixed a few typos.

* Removed "time series" as per review.

* Update docs/sources/http_api/data_source.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Changes from Diana's review.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

achatterjee-grafana committed:
addressed issues 28763 and 30314. (#31404)

thisisobate committed:
Cascader: updates story from knobs to controls (#31399)

zoltanbedi committed:
Chore: Upgrade eslint packages (#31408)

rubycut committed:
Add eu-south-1 cloudwatch region, closes #31197 (#31198)

thisisobate committed:
CallToActionCard: updates story from knobs to controls (#31393)

* CallToActionCard: updates story from knobs to controls

* used better naming for ctaElement control

* Capitalise Story names

* moved some changes to a separate PR

* fixes CI frontend build issue

domasx2 committed:
Logging: add frontend logging helpers to @grafana/runtime package (#30482)

relvira committed:
Add new Cloudwatch AWS/DDoSProtection metrics and dimensions (#31297)

electron0zero committed:
remove squadcast details from docs (#31413)

squadcast was removed from table in https://github.com/grafana/grafana/pull/23213 but we missed the details paragraph, remove that.

thisisobate committed:
Move NOOP_CONTROL to storybook utils and change to a standalone file (#31421)

dsotirakis committed:
Update grabpl from 0.5.38 to 0.5.42 version (#31419)

* Update grabpl from 0.5.41 to 0.5.42 version

ivanahuckova committed:
Devenv: Add gdev-influxdb2 data source (#31250)

* Add gdev-influxdb2 data source

* Change influxdb2 docker block to use 8087 port

* Update telegraf url

ivanahuckova committed:
Prometheus: Use configured HTTP method for /series and /labels endpoints (#31401)

* Run post-friendly request with set method first

* Improve messaging, retry only when post and specific status code

* Add comments

* Fix backend

* Update public/app/plugins/datasource/prometheus/datasource.ts

gabor committed:
InfluxDB: handle columns named "table" (#30985)

* influxdb: flux: manually calculate table-group-key

* influxdb: flux: added explanation about the group-key

dsotirakis committed:
Remove tidy-check script (#31423)

* Remove tidy-check script

* Update grabpl version to 0.5.43

bergquist committed:
add hg team as migrations code owners (#31420)

Signed-off-by: bergquist <carl.bergquist@gmail.com>

zoltanbedi committed:
TraceViewer: Fix trace to logs icon to show in right pane (#31414)

ilyastoli committed:
CloudWatch: Added AWS Ground Station metrics and dimensions (#31362)

ying-jeanne committed:
Postgres: allow providing TLS/SSL certificates as text in addition to file paths (#30353)

* postgres SSL certification

* add back the UI to configure SSL Authentication files by file path

* add backend logic

* correct unittest

* mini changes

* Update public/app/plugins/datasource/postgres/config_ctrl.ts

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update public/app/plugins/datasource/postgres/partials/config.html

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* mutex

* check file exist before remove

* change permission

* change default configuremethod to file-path

* Update public/app/plugins/datasource/postgres/partials/config.html

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Update public/app/plugins/datasource/postgres/partials/config.html

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Update public/app/plugins/datasource/postgres/partials/config.html

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Update public/app/plugins/datasource/postgres/partials/config.html

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* rename sslconfiguremethod to sslconfigurationmethod

* frontend update

* solve comments

* Postgres: Convert tests to stdlib

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Postgres: Be consistent about TLS/SSL terminology

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* fix init inconsistancy

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* naming convention

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo change

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix TLS issue

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* change permissions

* Fix data source field names

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Clean up HTML

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Improve popover text

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix SSL input bug

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo unnecessary change

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Clean up backend code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix build

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* More consistent naming

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Clean up code

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Enforce certificate file permissions

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* add settings

* Undo changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* fix windows file path

* PostgresDataSource: Fix mutex usage

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix tests

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/tsdb/postgres/postgres.go

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Apply suggestions from code review

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* fix compilation

* fix unittest

* Apply suggestions from code review

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Apply suggestions from code review

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* mock function

* change kmutex package

* add kmutex into middleware

* lock connection file per datasource

* add unittest regarding concurrency

* version should be equal

* adding unittest

* fix the loop

* fix unitest

* fix postgres unittst

* remove comments

* move dataPath from arg to tlsManager struct field

* Use DecryptedValues method

Use cached decrypted values instead of
using secure json data decrypt which will
decrypt unchanged values over and over again.

* remove unneeded mutex in tests and cleanup tests

* fix the lint

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>
Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

Commit:
Elasticsearch: Fix query initialization logic & query transformation from Promethous/Loki (#31322)

* Elasticsearch: Fix query initialization logic

* Only import prometheus & loki queries as log queries

ryantxu committed:
Update public/app/plugins/panel/timeseries/EditBanner.tsx

Co-authored-by: Hugo Häggmark <hugo.haggmark@grafana.com>

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu committed:
fix build

dprokop commented:
@ryantxu moving to 8.0. I was thinking about the same thing, but there is still few things we need to address before making this one a default one.

ryantxu committed:
merge master

ryantxu committed:
merge master

ryantxu committed:
merge master

osg-grafana commented:
Deferring to @achatterjee-grafana on any doc changes for this.

oddlittlebird commented:
Will the old Graph panel still be an option?

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu commented:
> Will the old Graph panel still be an option?

For sure existing graph panels will continue to work with flot in 8.  Ideally we can remove the ability to create new flot panels in 8 like we did with the single-stat >> stat migration.  BUT there is a long list of stuff that needs done before we can consider that!  See:  https://github.com/grafana/grafana/issues/30564

ryantxu committed:
show alert tab for timeseries

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=30919) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=30919) before we can accept your contribution.<br/>**2** out of **39** committers have signed the CLA.<br/><br/>:white_check_mark: gabor<br/>:white_check_mark: ivanahuckova<br/>:x: dafydd-t<br/>:x: peterholmberg<br/>:x: marefr<br/>:x: dprokop<br/>:x: aknuds1<br/>:x: kaydelaney<br/>:x: dupondje<br/>:x: jackw<br/>:x: sakjur<br/>:x: thisisobate<br/>:x: grafanabot<br/>:x: dsotirakis<br/>:x: matthewhelmke<br/>:x: aocenas<br/>:x: xlson<br/>:x: zoltanbedi<br/>:x: undef1nd<br/>:x: achatterjee-grafana<br/>:x: domasx2<br/>:x: hugohaggmark<br/>:x: electron0zero<br/>:x: Elfo404<br/>:x: joshhunt<br/>:x: Clarity-89<br/>:x: ying-jeanne<br/>:x: dependabot[bot]<br/>:x: wbrowne<br/>:x: torkelo<br/>:x: relvira<br/>:x: kurokochin<br/>:x: ryantxu<br/>:x: bergquist<br/>:x: LeviHarrison<br/>:x: vickyyyyyyy<br/>:x: oscarkilhed<br/>:x: rubycut<br/>:x: ilyastoli<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=30919) it.</sub>

ryantxu committed:
merge master

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu committed:
Merge remote-tracking branch 'origin/master' into 75-panel-defaults

ryantxu commented:
closing in favor of #33639
